toolchain/check/function.cpp

```diff
@@ -108,4 +127,39 @@ auto CheckFunctionReturnType(Context& context, SemIR::LocId loc_id,
   return return_info;
 }
 
+auto CheckFunctionDefinitionSignature(Context& context,
+                                      SemIR::FunctionId function_id) -> void {
+  auto& function = context.functions().Get(function_id);
+
+  auto params_to_complete =
+      context.inst_blocks().GetOrEmpty(function.call_params_id);
+
+  // Check the return type is complete.
+  if (function.return_slot_pattern_id.has_value()) {
+    CheckFunctionReturnType(context,
+                            SemIR::LocId(function.return_slot_pattern_id),
+                            function, SemIR::SpecificId::None);
+    params_to_complete = params_to_complete.drop_back();
```

Comment:
This was taking me a moment to see:
```suggestion
    // Don't re-check the return type below.
    params_to_complete = params_to_complete.drop_back();
```